core/src/main/java/org/apache/calcite/util/format/postgresql/ChronoUnitEnum.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.util.format.postgresql;
+
+import com.google.common.collect.ImmutableSet;
+
+import java.time.temporal.ChronoUnit;
+import java.util.Set;
+
+import static org.apache.calcite.util.format.postgresql.DateCalendarEnum.GREGORIAN;
+import static org.apache.calcite.util.format.postgresql.DateCalendarEnum.ISO_8601;
+import static org.apache.calcite.util.format.postgresql.DateCalendarEnum.JULIAN;
+import static org.apache.calcite.util.format.postgresql.DateCalendarEnum.NONE;
+
+/**
+ * A component of a datetime. May belong to a type of calendar. Also contains
+ * a list of parent values. For example months are in a year. A datetime can be
+ * reconstructed from one or more <code>ChronUnitEnum</code> items along with
+ * their associated values.
+ *
+ * <p>Some <code>ChronoUnitEnum</code> items conflict with others.
+ */
+public enum ChronoUnitEnum {
+  ERAS(ChronoUnit.ERAS, NONE),
+  CENTURIES(
+      ChronoUnit.CENTURIES,
+      ImmutableSet.of(ISO_8601, GREGORIAN),
+      ERAS),
+  YEARS_ISO_8601(
+      ChronoUnit.YEARS,
+      ISO_8601,
+      CENTURIES),
+  YEARS_IN_MILLENIA_ISO_8601(
+      ChronoUnit.YEARS,
+      ISO_8601,
+      CENTURIES),
+  YEARS_IN_CENTURY_ISO_8601(
+      ChronoUnit.YEARS,
+      ISO_8601,
+      CENTURIES),
+  DAYS_IN_YEAR_ISO_8601(
+      ChronoUnit.YEARS,
```

Comment:
is this unit right?

Reply:
It was not, good catch. This is corrected.